src/processes/poisson.jl

```diff
@@ -19,61 +24,94 @@ Base.union(p₁::PoissonProcess{<:Real}, p₂::PoissonProcess{<:Function}) = Poi
 
 Base.union(p₁::PoissonProcess{<:Function}, p₂::PoissonProcess{<:Real}) = PoissonProcess(x -> p₁.λ(x) + p₂.λ)
 
+Base.union(p₁::PoissonProcess{<:AbstractVector}, p₂::PoissonProcess{<:AbstractVector}) = PoissonProcess(x -> p₁.λ + p₂.λ)
+
 ishomogeneous(p::PoissonProcess{<:Real}) = true
 ishomogeneous(p::PoissonProcess{<:Function}) = false
-
-default_sampling_algorithm(::PoissonProcess, ::Geometry) = DiscretizedSampling()
-default_sampling_algorithm(::PoissonProcess{<:Real}, ::Box) = ProductSampling()
+ishomogeneous(p::PoissonProcess{<:Vector}) = false
+
+default_sampling_algorithm(::PoissonProcess) = DiscretizedSampling()
+default_sampling_algorithm(::PoissonProcess{<:Function}) = ThinnedSampling()
+
+default_lambda_max(p, g, algo::ThinnedSampling{T}) where {T<:Real} = algo.param
```

Comment:
Maybe this lambda_max is not a function of the algorithm, but some internal implementation detail that we can always give a good default depending on the domain? It feels strange to request a lambda_max from end-users. After all, all they need is to simulate a Poisson process with varying intensity lambda.